alicloud/service_alicloud_alikafka.go

```diff
@@ -0,0 +1,165 @@
+			return
+		}
+	}
+	return alikafkaInstance, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+}
+
+func (alikafkaService *AlikafkaService) DescribeAlikafkaConsumerGroup(id string) (alikafkaConsumerGroup *alikafka.ConsumerVO, err error) {
+
+	parts, err := ParseResourceId(id, 2)
+	if err != nil {
+		return nil, WrapError(err)
+	}
+	instanceId := parts[0]
+	consumerId := parts[1]
+
+	request := alikafka.CreateGetConsumerListRequest()
+	request.InstanceId = instanceId
+	request.RegionId = alikafkaService.client.RegionId
+
+	raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+		return alikafkaClient.GetConsumerList(request)
+	})
+
+	if err != nil {
+		return alikafkaConsumerGroup, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	consumerListResp, _ := raw.(*alikafka.GetConsumerListResponse)
+	addDebug(request.GetActionName(), raw)
+
+	for _, v := range consumerListResp.ConsumerList.ConsumerVO {
+		if v.ConsumerId == consumerId {
+			alikafkaConsumerGroup = &v
+			return
+		}
+	}
+	return alikafkaConsumerGroup, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+}
+
+func (alikafkaService *AlikafkaService) DescribeAlikafkaTopic(id string) (alikafkaTopic *alikafka.TopicVO, err error) {
+
+	parts, err := ParseResourceId(id, 2)
+	if err != nil {
+		return nil, WrapError(err)
+	}
+	instanceId := parts[0]
+	topic := parts[1]
+
+	request := alikafka.CreateGetTopicListRequest()
+	request.InstanceId = instanceId
+	request.RegionId = alikafkaService.client.RegionId
+
+	raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+		return alikafkaClient.GetTopicList(request)
+	})
+
+	if err != nil {
+		return alikafkaTopic, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	topicListResp, _ := raw.(*alikafka.GetTopicListResponse)
+	addDebug(request.GetActionName(), raw)
+
+	for _, v := range topicListResp.TopicList.TopicVO {
+		if v.Topic == topic {
+			alikafkaTopic = &v
+			return
+		}
+	}
+	return alikafkaTopic, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+}
+
+func (s *AlikafkaService) WaitForAlikafkaConsumerGroup(id string, status Status, timeout int) error {
+	deadline := time.Now().Add(time.Duration(timeout) * time.Second)
+	parts, err := ParseResourceId(id, 2)
+	if err != nil {
+		return WrapError(err)
+	}
+	instanceId := parts[0]
+	consumerId := parts[1]
+	for {
+		response, err := s.DescribeAlikafkaConsumerGroup(id)
+		if err != nil {
+			if NotFoundError(err) {
+				if status == Deleted {
+					return nil
+				}
+			} else {
+				return WrapError(err)
+			}
+		}
+
+		if response.InstanceId+":"+response.ConsumerId == id && status != Deleted {
+			return nil
+		}
+
+		if time.Now().After(deadline) {
+			return WrapErrorf(err, WaitTimeoutMsg, id, GetFunc(1), timeout, instanceId+":"+consumerId, id, ProviderERROR)
```

Comment:
instanceId+":"+consumerId    should be    object.InstanceId+":"+ object.ConsumerId